src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -708,19 +708,33 @@ class ColumnReferenceSegment(ObjectReferenceSegment):  # type: ignore
     )
 
     def extract_possible_references(self, level):
-        """Extract possible references of a given level."""
+        """Extract possible references of a given level.
+
+        Overrides the parent-class function. BigQuery's support for things like
+        the following
+        - Functions that take a table as a parameter (e.g. TO_JSON_STRING)
+          https://cloud.google.com/bigquery/docs/reference/standard-sql/
+          json_functions#to_json_string
+        - STRUCT
+
+        mean that, without schema information (which SQLFluff does not have),
+        references to data are often ambiguous.
+        """
         level = self._level_to_int(level)
         refs = list(self.iter_raw_references())
         if level == self.ObjectReferenceLevel.SCHEMA.value and len(refs) >= 3:
             return [refs[0]]  # pragma: no cover
-        if level == self.ObjectReferenceLevel.TABLE.value and len(refs) >= 3:
-            # Ambiguous case: The table could be the first or second part, so
-            # return both.
-            return [refs[0], refs[1]]
-        if level == self.ObjectReferenceLevel.OBJECT.value and len(refs) >= 3:
+        if level == self.ObjectReferenceLevel.TABLE.value:
+            # One part: Could be a table, e.g. TO_JSON_STRING(t)
+            # Two parts: Could be schema+table or table+column.
+            # Three parts: Could be table+column+struct or schema+table+column.
```

Comment:
```suggestion
            # Three parts: Could be table+column+struct or dataset+table+column.
```

In theory there could be four parts (dataset.table.column.struct), or even five parts (project.dataset.table.column.struct) in which case the current query won’t pick this up.

Similarly I think it could be several structs (e.g. table.struct.struct).

I wonder is it better to look at first three parts instead of last two? That should always get the table.

Reply:
It's actually looking at the first 2, not the last 2. Will change it to the first 3, though. Good catch! 👍

Reply:
Actually I’m having second thoughts about this now…

This rule only cares about alias’s that are unused. Therefore my suggestions of 4 and 5 aren’t valid as those are full table references and not alias references. So think you should revert back to what you originally had.

In fact is first part sufficient instead of first two parts?

Reply:
I think your point was valid, in the sense that this function (on the BigQuery dialect side) deals with general object references -- it doesn't know that the caller is only interested in aliases. So I think this is reasonable for now, unless we want to get super precise and distinguish between callers asking for TABLE vs ALIAS. We _could_ do that, but it feels a bit like overengineering at this point, maybe?

Reply:
Ah fair point. This function potentially isn’t just for L025. Ok fine to leave as is then with three.